rclcpp_action/include/rclcpp_action/client.hpp

```diff
@@ -410,10 +410,7 @@ class Client : public ClientBase
       // This will override any previously registered callback
       goal_handle->set_result_callback(result_callback);
     }
-    // If the user chose to ignore the result before, then ask the server for the result now.
-    if (!goal_handle->is_result_aware()) {
```

Comment:
There was a race between this line and the next; the goal handle could be made "aware" between this condition and setting awareness below.
This has been fixed by the guard in `make_result_aware` below.